File: core/src/test/java/net/dontdrinkandroot/wicket/behavior/TagNameModifierTest.java
package net.dontdrinkandroot.wicket.behavior;

import net.dontdrinkandroot.wicket.test.AbstractWicketTest;
import org.apache.wicket.core.util.string.ComponentRenderer;
import org.apache.wicket.markup.html.WebMarkupContainer;
import org.junit.Assert;
import org.junit.Test;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class TagNameModifierTest extends AbstractWicketTest
{
    @Test
    public void testModelConstructor()
    {
        WebMarkupContainer container = new WebMarkupContainer("id");
        container.add(new TagNameModifier("div"));
        CharSequence componentMarkup = ComponentRenderer.renderComponent(container);

        Assert.assertEquals(
                "<wicket:div wicket:id=\"id\"></wicket:div>",
                componentMarkup.toString()
        );
    }
}

File: core/src/test/kotlin/net/dontdrinkandroot/wicket/behavior/TagNameModifierTest.kt
package net.dontdrinkandroot.wicket.behavior

import net.dontdrinkandroot.wicket.test.AbstractWicketTest
import org.apache.wicket.core.util.string.ComponentRenderer
import org.apache.wicket.markup.html.WebMarkupContainer
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class TagNameModifierTest : AbstractWicketTest() {

    @Test
    fun testModelConstructor() {
        val container = WebMarkupContainer("id")
        container.add(TagNameModifier("div"))
        val componentMarkup = ComponentRenderer.renderComponent(container)
        Assertions.assertEquals(
            """<wicket:div wicket:id="id"></wicket:div>""",
            componentMarkup.toString()
        )
    }
}